src/Util.cpp

```diff
@@ -572,6 +577,63 @@ int get_llvm_version() {
     return LLVM_VERSION;
 }
 
+#ifdef _WIN32
+
+namespace {
+
+struct DeferredFunction {
+    const std::function<void()> *run;
+    LPVOID fiber;
+};
+
+void generic_fiber_entry_point(void *argument) {
+    auto *action = reinterpret_cast<DeferredFunction *>(argument);
+    (*action->run)();
+    SwitchToFiber(action->fiber);
+}
+
+}  // namespace
+
+#endif
+
+void run_with_large_stack(const std::function<void()> &action) {
+#if _WIN32
+    SIZE_T required_stack = 8 * 1024 * 1024;
+
+    ULONG_PTR stack_low, stack_high;
+    GetCurrentThreadStackLimits(&stack_low, &stack_high);
+    ptrdiff_t stack_remaining = (char *)&stack_high - (char *)stack_low;
```

Comment:
Heh, this is slightly too clever for me. Would be better to make a local variable explicitly for the purpose of taking its address instead of reusing an existing one.

Reply:
Fixed :)